src/Client/DefaultRegistrationService.php

```diff
@@ -57,6 +57,10 @@ public function register(iterable $strategyHandlers): bool
                 'started' => (new DateTimeImmutable())->format('c'),
                 'interval' => $this->configuration->getMetricsInterval(),
             ], JSON_THROW_ON_ERROR)));
+        
+        if ($this->configuration->getProxyKey() !== null) {
+            $request = $request->withHeader('Authorization', $this->configuration->getProxyKey());
+        }
```

Comment:
Well, this is rather unfortunate solution, but sadly I don't think we have any other quick fix. The proxy key should be directly tied to the headers in configuration.

Can you add a comment with `TODO: refactor proxy key into headers` or something similar, so that I'll pick it up next time I'm doing clean-up?

Reply:
I can have another stab before converting this to a proper PR.

Reply:
Totally agree that this should be used when we build the configuration.